lib/biz/time.rb

```diff
@@ -39,7 +40,12 @@ def on_date(date, day_time)
         true
       )
     rescue TZInfo::PeriodNotFound
-      on_date(date, DayTime.new(day_time.day_second + HOUR))
+      next_day_second = day_time.day_second + HOUR
+      if next_day_second > SECONDS_IN_DAY
+        on_date(date + 1, DayTime.new(next_day_second % SECONDS_IN_DAY))
```

Comment:
yeah, wasn't sure whether this could be simplified to drop the modulus, but this is undoubtedly the safest :)